tor/cmd_onion.go

```diff
@@ -3,88 +3,30 @@ package tor
 import (
 	"errors"
 	"fmt"
-	"io/ioutil"
-	"os"
-)
-
-var (
-	// ErrNoPrivateKey is an error returned by the OnionStore.PrivateKey
-	// method when a private key hasn't yet been stored.
-	ErrNoPrivateKey = errors.New("private key not found")
-)
 
-// OnionType denotes the type of the onion service.
-type OnionType int
-
-const (
-	// V2 denotes that the onion service is V2.
-	V2 OnionType = iota
-
-	// V3 denotes that the onion service is V3.
-	V3
+	"github.com/lightningnetwork/lnd/tor/onionfile"
 )
 
 // OnionStore is a store containing information about a particular onion
 // service.
 type OnionStore interface {
 	// StorePrivateKey stores the private key according to the
 	// implementation of the OnionStore interface.
-	StorePrivateKey(OnionType, []byte) error
+	StorePrivateKey([]byte) error
 
 	// PrivateKey retrieves a stored private key. If it is not found, then
 	// ErrNoPrivateKey should be returned.
-	PrivateKey(OnionType) ([]byte, error)
+	PrivateKey() ([]byte, error)
 
 	// DeletePrivateKey securely removes the private key from the store.
-	DeletePrivateKey(OnionType) error
-}
-
-// OnionFile is a file-based implementation of the OnionStore interface that
-// stores an onion service's private key.
-type OnionFile struct {
-	privateKeyPath string
-	privateKeyPerm os.FileMode
-}
-
-// A compile-time constraint to ensure OnionFile satisfies the OnionStore
-// interface.
-var _ OnionStore = (*OnionFile)(nil)
-
-// NewOnionFile creates a file-based implementation of the OnionStore interface
-// to store an onion service's private key.
-func NewOnionFile(privateKeyPath string,
-	privateKeyPerm os.FileMode) *OnionFile {
-
-	return &OnionFile{
-		privateKeyPath: privateKeyPath,
-		privateKeyPerm: privateKeyPerm,
-	}
-}
-
-// StorePrivateKey stores the private key at its expected path.
-func (f *OnionFile) StorePrivateKey(_ OnionType, privateKey []byte) error {
-	return ioutil.WriteFile(f.privateKeyPath, privateKey, f.privateKeyPerm)
-}
-
-// PrivateKey retrieves the private key from its expected path. If the file
-// does not exist, then ErrNoPrivateKey is returned.
-func (f *OnionFile) PrivateKey(_ OnionType) ([]byte, error) {
-	if _, err := os.Stat(f.privateKeyPath); os.IsNotExist(err) {
-		return nil, ErrNoPrivateKey
-	}
-	return ioutil.ReadFile(f.privateKeyPath)
-}
-
-// DeletePrivateKey removes the file containing the private key.
-func (f *OnionFile) DeletePrivateKey(_ OnionType) error {
-	return os.Remove(f.privateKeyPath)
+	DeletePrivateKey() error
 }
 
 // AddOnionConfig houses all of the required parameters in order to
 // successfully create a new onion service or restore an existing one.
 type AddOnionConfig struct {
```

Comment:
You should be able to remove the exported `OnionStore` now and replace it with `onionStore`. If I'm wrong then you can just get rid of the unexported `onionStore`